src/CMakeLists.txt

```diff
@@ -7,7 +7,7 @@ cmake_minimum_required(VERSION 3.10)
 project(esp)
 
 if(MSVC)
-  add_definitions(/DNOMINMAX)
+  add_definitions(/DNOMINMAX  /D_USE_MATH_DEFINES /DEIGEN_DONT_ALIGN_STATICALLY)
```

Comment:
What is `_USE_MATH_DEFINES` needed for? Does Habitat code need it or rather some dependency? If Habitat, we have our own constants for that, if 3rd party code, I'd prefere to have that specified in `src/deps/CMakeLists.txt` instead.

Reply:
Two projects (nav and continuous) fail to build because M_PI is undefined.

Reply:
@cegbertOculus if sophus is the only culprit, perhaps we can have the define go along with the inclusion of sophus at https://github.com/facebookresearch/habitat-sim/blob/master/src/cmake/dependencies.cmake#L26

Reply:
From what I can tell, the inclusion of sophus in dependencies.cmake just specifies where to include the sophus files from, and I can't define anything with that inclusion.  Maybe it would make sense to put it in src/cmakelists.txt where the projects are added?  Like this: add_subdirectory(esp/nav)
target_compile_definitions(nav PUBLIC _USE_MATH_DEFINES )

Reply:
Ah, you're right, I believe.  Yes, it should be fine to add conditional target defines as above.  Perhaps also with a short comment above the block stating that this is done for the sophus dependency, just for readability

Reply:
Looking at Sophus' own CMakeLists, it sets it [implicitly for the sophus target](https://github.com/strasdat/Sophus/blob/05f9645077c4e59ca9aa05bb0d0eaf0dd83d2b26/CMakeLists.txt#L68) -- so I think the cleanest option is to explicitly link to the `sophus` target instead of relying just on it being inside the [global include path](https://github.com/facebookresearch/habitat-sim/blob/bb87a9f1626d6f7d25a2513bf8dcb85123b9e5e2/src/cmake/dependencies.cmake#L26)? Like

```cmake
target_link_libraries(nav 
  PUBLIC 
    ... 
    sophus)
```

and similar for `gfx` and `assets`. Then we could even remove the global include path for Sophus, I think -- in any case I'm planning to do that as part of #42.

Reply:
Explicitly linking didn't work, so I've just added:

```
if(MSVC)
  # Define common math constants if we compile with MSVC
  target_compile_definitions(assets INTERFACE _USE_MATH_DEFINES)
endif (MSVC)
```

to the cmakelists.txt for nav, gfx, and assets.  Is that acceptable?

Reply:
`PUBLIC` instead of `PRIVATE`? I assume those definitions are needed when compiling the library itself as well. And please mention it's because of Sophus, otherwise this knowledge gets lost and somebody might remove it again in the future.